### PR TITLE
Support top level declarations with templates

### PR DESCRIPTION
**Pull request by keertip**

### CODE REVIEW

**Comment by keertip** [comment]
fixes #1980

File: lib/src/model.dart [hunk 1]
```diff
@@ -4911,6 +4911,10 @@ class PackageGraph {
         if (needsPrecacheRegExp.hasMatch(d.documentationComment) &&
             !precachedElements.contains(d)) {
           precachedElements.add(d);
+          if (m is TopLevelVariable) {
```
Comment (devoncarew) [inline lib/src/model.dart]: Can you add a comment explaining why `TopLevelVariable`s are special cased here?

**Comment by devoncarew** [comment]
> fixes #1980

🎉 great to have the build fixed!

File: lib/src/model.dart (hunk 1 above)
Comment (keertip) [reply lib/src/model.dart]: done

**Pull request merged**